clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2079,7 +2079,9 @@ class SyclKernelFieldChecker : public SyclKernelFieldHandler {
   }
 
   bool handleSyclSpecialType(ParmVarDecl *PD, QualType ParamTy) final {
-    if (!SemaSYCL::isSyclType(ParamTy, SYCLTypeAttr::work_group_memory)) {
+    if (!SemaSYCL::isSyclType(ParamTy, SYCLTypeAttr::work_group_memory) &&
```

Comment:
Could you please add clang LIT tests for the change? grep ""work_group_memory" in clang/test directory for examples.

Reply:
Okay, I'll add such tests.

Reply:
Done.